tests/e2e_tests/test_bp_map.py

```diff
@@ -0,0 +1,255 @@
+        master_dark.save(filedir=bp_map_outputdir, filename = "dark_mock.fits")
+        this_caldb.create_entry(master_dark)
+        master_dark_ref = master_dark.filepath
+
+    ####### Run the walker
+    walker.walk_corgidrp(input_image_filelist, "", bp_map_outputdir, template="bp_map.json")
+
+    # Clean up by removing entry
+    this_caldb.remove_entry(noise_map)
+    this_caldb.remove_entry(flat)
+    this_caldb.remove_entry(master_dark)
+
+    generated_bp_map_file = os.path.join(bp_map_outputdir, "dark_mock_bad_pixel_map.fits")
+
+    # Test generated BP map against TVAC data or simple dark
+    generated_bp_map_img = data.Image(generated_bp_map_file)
+
+    if use_master_dark:
+        with fits.open(master_dark_ref) as hdulist:
+            dark_ref_dat = hdulist[1].data
+
+            # Check for bad pixels
+            bad_pixels = generated_bp_map_img.data[generated_bp_map_img.data > 0]
+            if bad_pixels.size > 0:
+                print(f"Bad pixel values identified: {bad_pixels}")
+            else:
+                print("No bad pixels identified")
+            fig, axes = plt.subplots(2, 3, figsize=(15, 10))
+            # Flatten the axes array for easier indexing
+            axes = axes.flatten()
+
+            # First subplot
+            im1 = axes[0].imshow(fpn_dat, vmin=-240, vmax=240, cmap="gray")
+            axes[0].set_title("FPN noise map")
+            axes[0].set_xlim([0, naxis1])
+            axes[0].set_ylim([0, naxis2])
+            fig.colorbar(im1, ax=axes[0])
+
+            # Second subplot
+            im2 = axes[1].imshow(cic_dat, vmin=0, vmax=0.02, cmap="gray")
+            axes[1].set_title("CIC noise map")
+            axes[1].set_xlim([0, naxis1])
+            axes[1].set_ylim([0, naxis2])
+            fig.colorbar(im2, ax=axes[1])
+
+            # Third subplot
+            im3 = axes[2].imshow(dark_current_dat, vmin=0, vmax=0.003, cmap="gray")
+            axes[2].set_title("Dark current noise map")
+            axes[2].set_xlim([0, naxis1])
+            axes[2].set_ylim([0, naxis2])
+            fig.colorbar(im3, ax=axes[2])
+
+            # Fourth subplot (on the second row, left)
+            im4 = axes[3].imshow(dark_ref_dat, vmin=-240, vmax=240, cmap="gray")
+            axes[3].set_title("DRP-produced master dark")
+            axes[3].set_xlim([0, naxis1])
+            axes[3].set_ylim([0, naxis2])
+            fig.colorbar(im4, ax=axes[3])
+
+            # Fifth subplot (on the second row, right)
+            im5 = axes[4].imshow(generated_bp_map_img.data, vmin=0, vmax=8, cmap="gray")
+            axes[4].set_title("DRP-produced BP map")
+            axes[4].set_xlim([0, naxis1])
+            axes[4].set_ylim([0, naxis2])
+            fig.colorbar(im5, ax=axes[4])
+
+            # Hide the unused subplot (bottom-right corner)
+            axes[5].axis('off')
+
+            plt.tight_layout()
+            plt.show()
```

Comment:
Will comment out. It may be useful to save the image to disk if we ever get a BP map with any identified bad pixels, but can uncomment this block of code if so.

Reply:
You can do `plt.savefig()` if you want but we shouldn't use `plot.show()` because it's a blocking operation that disrupts automation.

Reply:
Done